### ISSUE TITLE
IR

### ISSUE THREAD

**Issue by soundsnow** [philj56/gbcc#34]
The Game Boy Color had an infrared transmitter and receiver. Generation II Pokémon games made use of this feature through Mystery Gift. Super Mario Bros. Deluxe could send high scores to another cartridge.
full IR emulation for most IR-enabled games (Hudson Soft's HuC-1's IR capabilities need to be emulated for Japanese Pokémon TCG).

**Comment by philj56** [comment]
Yup, IR is something I just haven't attempted yet (and is kind of pointless until I get multiplayer working anyway)

**Comment by 360fun** [comment]
This feature has so much potential for ROM development, way beyond playing old games with IR support! 🤩
Imagine enabling wireless IR communication between an Android phone (many Xiaomi devices have IR blasters) and a Game Boy Color with a custom ROM (flash carts are very cheap nowadays):

```
GBC <=IR=> ANDROID <=WIFI/CELL=> INTERNET  
GBC <=IR=> ANDROID <=WIFI=> LOCAL NETWORK  
GBC <=IR=> ANDROID <=BT=> IoT DEVICES / CONTROLLERS...
```

We could develop custom GBC ROMs that interact with the smartphone—sending and receiving data through IR. This could unlock incredible new use cases:
	•	GBC as a retro-style IoT interface
	•	Online multiplayer games
	•	Real-time data-driven games and apps
	•	and much more...

It would bridge retro hardware with modern tech! 🦄
